### PR TITLE
API allows user to sign up for an account and request a community

### PR DESCRIPTION
**Pull request by pgaunitz**
### User Story
```
As a user
In order to be part of a community
I would like to be able to create an account with a community request
```
## Changes proposed in this pull request:
* Adds community model
* Adds community controller
* Adds community status to user, pending, accepted, rejected
* Adds association between community and user
## What I have learned working on this feature:
* How generate random alphanumeric code using SecureRandom
* Learned to visualize data flow in the app

### CODE REVIEW

File: db/migrate/20200415090814_create_communities.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class CreateCommunities < ActiveRecord::Migration[6.0]
+  def change
+    create_table :communities do |t|
+      t.string :code
+      t.string :name
+
```
Comment (kfrostare) [inline db/migrate/20200415090814_create_communities.rb:6]: Is the space supposed to be there? If yes then ... don't mind this comment :P

File: spec/factories/users.rb [hunk 2]
```diff
@@ -1,13 +1,17 @@
+# frozen_string_literal: true
```
Comment (kfrostare) [inline spec/factories/users.rb:1]: Should this commented out line be there? If yes ignore this comment :-P

File: spec/models/community_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe Community, type: :model do
+  it 'should have valid Factory' do
+    expect(create(:community)).to be_valid
+  end
+
```
Comment (kfrostare) [inline spec/models/community_spec.rb:7]: Should this space be here and if yes, why should it not also be between 12 and 13?

Comment (pgaunitz) [reply spec/models/community_spec.rb:7]: fixed

File: spec/requests/user_can_send_request_to_community_spec.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+RSpec.describe 'GET /communities' do
+  let!(:community) { create(:community, code: '2tUnJLGwPYJq') }
+  let!(:community2) { create(:community, code: 'paYljxCDcjGE') }
+  let!(:community3) { create(:community, code: 'Hqxx5Uk5evmD') }
+
```
Comment (kfrostare) [inline spec/requests/user_can_send_request_to_community_spec.rb:5]: In this file some blocks have spaces between them and some does not, just a question of consistency. /Kf

File: spec/requests/session_spec.rb [hunk 5]
```diff
@@ -7,6 +7,7 @@
     { HTTP_ACCEPT: 'application/json' }.merge!(admin_credentials)
   end
 
+  # let(:community) { create(:community) }
```
Comment (kaylawoodbury) [inline spec/requests/session_spec.rb]: do we need this comment?

**Review by kaylawoodbury — APPROVED** [review]
Think there is one comment that can be removed, otherwise I think this looks good!

**Pull request merged**